user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/Ad_hoc_data/Ad_hoc_Tutorials_Satellites.md

```diff
@@ -0,0 +1,89 @@
+
+Expected duration: 15 minutes.
+
+## Prerequisites
+
+- [Visual Studio](https://visualstudio.microsoft.com/downloads/) with [DataMiner Integration Studio](xref:Installing_and_configuring_DataMiner_Integration_Studio)
+- A DataMiner Agent [connected to dataminer.services](xref:Connecting_your_DataMiner_System_to_the_cloud)
+
+## Overview
+
+- [Step 1: Get a quick start from the catalog](#step-1-get-a-quick-start-from-the-catalog)
+- [Step 2: Open the data source in Visual Studio](#step-2-open-the-data-source-in-visual-studio)
+- [Step 3: Complete the data source](#step-3-complete-the-data-source)
+- [Step 4: Use the data source](#step-4-use-the-data-source)
+
+## Step 1: Get a quick start from the catalog
+
+1. Go to [https://catalog.dataminer.services/catalog/5406](https://catalog.dataminer.services/catalog/5406)
+
+1. Deploy the catalog item to your DataMiner Agent.
+
+## Step 2: Open the data source in Visual Studio
+
+1. Open Visual Studio and select *Extensions* > *DIS* > *DMA* > *Connect* to connect DIS to your DMA.
+
+1. Select *Extensions* > *DIS* > *DMA* > *Import Automation Script*.
+
+1. Select *Satellites - GQIDS* and click *Import*.
+
+1. Go to the C# code of the imported Automation script.
+
+## Step 3: Complete the data source
+
+1. Add the *IGQIDataSource* interface to the *SatellitesDataSource* class.
+
+1. Implement the *GetColumns* method to provide GQI with all the columns of the data source.
+
+   ```csharp
+       public GQIColumn[] GetColumns()
+       {
+           return new GQIColumn[] {
+               new GQIStringColumn("Name"),
+               new GQIDoubleColumn("Latitude"),
+               new GQIDoubleColumn("Longitude")
+           };
+       }
+   ```
+
+1. Implement the *GetNextPage* method to provide GQI with the actual data. Each row is one satellite.
+
+   ```csharp
+       public GQIPage GetNextPage(GetNextPageInputArgs args)
+       {
+           var satellites = _satellitesHelper.GetSatellites().Take(100);
+           var rows = new List<GQIRow>();
+           foreach (var satellite in satellites) {
+               var cells = new GQICell[] {
+                   new GQICell(){ Value = satellite.Name },
+                   new GQICell(){ Value = satellite.Latitude},
+                   new GQICell() { Value = satellite.Longitude}
+               };
+               var row = new GQIRow(cells);
+               rows.Add(row);
+           }
+
+           return new GQIPage(rows.ToArray());
+       }
+   ```
+
+   By implementing both methods provided by the *IGQIDataSource* interface, you have now successfully finished the data source.
+
+1. To publish your changes, go to the XML file and click *Publish* in the top left corner.
+
+## Step 4: Use the data source
+
+1. Open the *Dashboards* app, [create a new dashboard](xref:Creating_a_completely_new_dashboard) and go to *queries* in the right panel.
```

Comment:
This is a bit too much for one step. Could you split this up into several steps and add a screenshot for the last one?